### PR TITLE
Cloud Manager 2022.6.0 API Docs

### PR DESCRIPTION
**Pull request by adtanase-adobe**
Cloud Manager 2022.6.0 API Docs

### CODE REVIEW

File: swagger-specs/api.yaml [hunk 1]
```diff
@@ -1374,6 +1375,238 @@ paths:
           description: Successful setting of variables
         '404':
           description: Environment not found
+  '/api/program/{programId}/environment/{environmentId}/restorePoints':
+    get:
+      tags:
+        - Environment Restore From Backup
+      summary: Get Restore Points
+      description: List the Backups available for this environment. (Cloud Service only).
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:1383]: Backup should not be capitalized here

File: swagger-specs/api.yaml [hunk 2]
```diff
@@ -1374,6 +1375,238 @@ paths:
...
+          required: true
+          type: string
+        - name: environmentId
+          in: path
+          description: Identifier of the environment
+          required: true
+          type: string
+        - name: x-gw-ims-org-id
+          in: header
+          description: IMS organization ID that the request is being made under.
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:1398]: Not a sentence, please don't end with a period/full stop

File: swagger-specs/api.yaml [hunk 3]
```diff
@@ -1374,6 +1375,238 @@ paths:
...
+          schema:
+            $ref: '#/definitions/RestorePointsList'
+        '404':
+          description: Resource not found
+  '/api/program/{programId}/environment/{environmentId}/restoreExecution/{restoreId}':
+    get:
+      tags:
+        - Environment Restore From Backup
+      summary: Get Restore Execution
+      description: Returns a restore execution by its id. (Cloud Service only).
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:1423]: Remove periods/full stops from sentence fragments

File: swagger-specs/api.yaml [hunk 4]
```diff
@@ -1374,6 +1375,238 @@ paths:
...
+          required: true
+          type: string
+        - name: restoreId
+          in: path
+          description: Identifier of the restore execution
+          required: true
+          type: string
+        - name: x-gw-ims-org-id
+          in: header
+          description: IMS organization ID that the request is being made under.
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:1443]: Remove period/full stop from sentence fragment

File: swagger-specs/api.yaml [hunk 5]
```diff
@@ -1374,6 +1375,238 @@ paths:
...
+        '200':
+          description: successful operation
+          schema:
+            $ref: '#/definitions/RestoreExecution'
+  '/api/program/{programId}/environment/{environmentId}/restoreExecution/{restoreId}/logs':
+    get:
+      tags:
+        - Environment Restore From Backup
+      summary: Get Restore Execution Logs
+      description: Gets the logs of a Restore Execution. (Cloud Service only).
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:1466]: Remove period/full stop from sentence fragment

File: swagger-specs/api.yaml [hunk 6]
```diff
@@ -1374,6 +1375,238 @@ paths:
...
+          type: string
+      responses:
+        default:
+          description: successful operation
+  '/api/program/{programId}/environment/{environmentId}/restoreExecutions':
+    get:
+      tags:
+        - Environment Restore From Backup
+      summary: Lists Restore Executions
+      description: List the Restore Executions of an environment. (Cloud Service only).
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:1511]: Remove period/full stop from sentence fragments

File: swagger-specs/api.yaml [hunk 7]
```diff
@@ -1374,6 +1375,238 @@ paths:
...
+        - name: limit
+          in: query
+          description: Pagination limit parameter
+          required: false
+          type: integer
+          default: 5
+          format: int32
+        - name: x-gw-ims-org-id
+          in: header
+          description: IMS organization ID that the request is being made under.
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:1539]: Remove period/full stop from sentence fragment

File: swagger-specs/api.yaml [hunk 8]
```diff
@@ -1374,6 +1375,238 @@ paths:
...
+        '200':
+          description: successful operation
+          schema:
+            $ref: '#/definitions/RestoreExecutionList'
+  '/api/program/{programId}/environment/{environmentId}/restoreExecution':
+    put:
+      tags:
+        - Environment Restore From Backup
+      summary: Start a Restore Execution
+      description: Start a Restore from Backup execution. (Cloud Service only).
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:1562]: Remove period/full stop from sentence fragments

File: swagger-specs/api.yaml [hunk 9]
```diff
@@ -1374,6 +1375,238 @@ paths:
...
+          type: string
+        - in: body
+          name: body
+          description: Restore Parameters Representation
+          required: false
+          schema:
+            $ref: '#/definitions/RestoreParamRepresentation'
+        - name: x-gw-ims-org-id
+          in: header
+          description: IMS organization ID that the request is being made under.
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:1583]: Remove period/full stop from sentence fragment

File: swagger-specs/api.yaml [hunk 10]
```diff
@@ -2481,6 +2714,124 @@ paths:
...
+      operationId: getNewRelicSubAccountUserList
+      parameters:
+        - name: programId
+          in: path
+          description: Identifier of the program
+          required: true
+          type: string
+        - name: x-gw-ims-org-id
+          in: header
+          description: IMS organization ID that the request is being made under.
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:2732]: Remove period/full stop from sentence fragment

File: swagger-specs/api.yaml [hunk 11]
```diff
@@ -2481,6 +2714,124 @@ paths:
...
+          required: true
+          type: string
+        - name: x-api-key
+          in: header
+          description: IMS Client ID (API Key) which is subscribed to consume services on console.adobe.io
+          required: true
+          type: string
+      responses:
+        '200':
+          description: Successful retrieval of the list of New Relic Sub Account users
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:2747]: "Sub account" is not a proper noun, do not capitalize

File: swagger-specs/api.yaml [hunk 12]
```diff
@@ -2481,6 +2714,124 @@ paths:
...
+        '401':
+          description: The request has not been applied because it lacks valid authentication credentials for the target resource.
+        '403':
+          description: The requester is not authorized to access the resource.
+        '405':
+          description: Client sent a request using a HTTP method that the server doesn't support.
+        '406':
+          description: Unacceptable content type. Client sent an Accept header for a content type which does not exist on the server.
+        '422':
+          description: Unprocessable Entity.
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:2761]: Remove period/full stop from sentence fragment

File: swagger-specs/api.yaml [hunk 13]
```diff
@@ -2481,6 +2714,124 @@ paths:
...
+          description: This is an indicator that the back-end service did not send a valid response.
+        '503':
+          description: This is an indicator of a potential server overload.
+        '504':
+          description: This is an indicator that the back-end service did not complete a response within an allowed time period.
+    patch:
+      tags:
+        - Programs
+      summary: Create and Delete New Relic Sub Account users
+      description: Create and Delete New Relic Sub Account users for this program
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:2774]: "Sub account" is not a proper noun and should not be capitalized.

File: swagger-specs/api.yaml [hunk 14]
```diff
@@ -4122,6 +4477,240 @@ definitions:
...
+            description: restorePointsDetails
+            items:
+              $ref: '#/definitions/RestorePointDetails'
+      _links:
+        type: object
+        readOnly: true
+        properties:
+          next:
+            $ref: '#/definitions/HalLink'
+            description: The next page of results.
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:4553]: Drop period/full stop from sentence fragment

File: swagger-specs/api.yaml [hunk 15]
```diff
@@ -4122,6 +4477,240 @@ definitions:
...
+        example: not_started
+        description: Status of the restore execution
+        enum:
+          - not_started
+          - running
+          - finished
+          - error
+          - failed
+      errorDetails:
+        description: Error details. Present only when execution failed.
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:4620]: Incomplete sentences, no need for periods, for example "Error details, present only when execution failed".

File: swagger-specs/api.yaml [hunk 16]
```diff
@@ -4122,6 +4477,240 @@ definitions:
...
+        properties:
+          next:
+            $ref: '#/definitions/HalLink'
+            description: The next page of results.
+          page:
+            $ref: '#/definitions/HalLink'
+            description: An arbitrary page of results.
+          prev:
+            $ref: '#/definitions/HalLink'
+            description: The previous page of results.
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:4701]: Drop period/full stop for sentence fragment

File: swagger-specs/api.yaml [hunk 17]
```diff
@@ -4515,6 +5104,67 @@ definitions:
           self:
             $ref: '#/definitions/HalLink'
             description: This object's representation
+  NewRelicSub-AccountUser:
+    type: object
+    properties:
+      id:
+        type: string
+        example: '42'
+        description: Identifier of the New Relic Sub Account User
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:5113]: Sub account is not a proper noun, do not capitalize

File: swagger-specs/api.yaml [hunk 18]
```diff
@@ -4515,6 +5104,67 @@ definitions:
...
+      firstName:
+        type: string
+        example: John
+        description: First name of the New Relic Sub Account User
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:5117]: Sub account is not a proper noun, do not capitalize

File: swagger-specs/api.yaml [hunk 19]
```diff
@@ -4515,6 +5104,67 @@ definitions:
...
+      lastName:
+        type: string
+        example: Doe
+        description: Last name of the New Relic Sub Account User
+      email:
+        type: string
+        example: foobar@newrelic.com
+        description: email of the New Relic Sub Account user
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:5125]: Sub account is not a proper noun, do not capitalize

File: swagger-specs/api.yaml [hunk 20]
```diff
@@ -4515,6 +5104,67 @@ definitions:
...
+      role:
+        type: string
+        example: admin
+        description: Role of the New Relic Sub Account user
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:5129]: Sub account is not a proper noun, do not capitalize

File: swagger-specs/api.yaml [hunk 21]
```diff
@@ -4515,6 +5104,67 @@ definitions:
...
+      owner:
+        type: boolean
+        example: false
+        description: Owner detail of the New Relic Sub Account User
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:5133]: Sub account is not a proper noun, do not capitalize

File: swagger-specs/api.yaml [hunk 22]
```diff
@@ -4515,6 +5104,67 @@ definitions:
...
+  NewRelicSubAccountUserlist:
+    type: object
+    properties:
+      _totalNumberOfNewRelicSubAccountUsers:
+        type: integer
+        format: int32
+      applicationId:
+        type: string
+        example: '44'
+        description: The id of the Cloud Manager application this sub-account user list is linked to
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:5145]: Sub account was not hyphenated previously.

File: swagger-specs/api.yaml [hunk 23]
```diff
@@ -4515,6 +5104,67 @@ definitions:
...
+          newRelicUsers:
+            type: array
+            items:
+              $ref: '#/definitions/NewRelicSub-AccountUser'
+      _links:
+        type: object
+        readOnly: true
+        properties:
+          'http://ns.adobe.com/adobecloud/rel/application':
+            description: Application linked to this New Relic Sub Account
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:5159]: Sub account is not a proper noun, do not capitalize

File: swagger-specs/api.yaml [hunk 24]
```diff
@@ -4515,6 +5104,67 @@ definitions:
...
+            $ref: '#/definitions/HalLink'
+          'http://ns.adobe.com/adobecloud/rel/newRelicSubAccount':
+            description: New Relic Sub Account linked to the New Relic Sub Account User list
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:5162]: Sub account is not a proper noun, do not capitalize

File: swagger-specs/api.yaml [hunk 25]
```diff
@@ -4515,6 +5104,67 @@ definitions:
...
+            $ref: '#/definitions/HalLink'
+          self:
+            $ref: '#/definitions/HalLink'
+            description: This object's representation
+    description: List of New Relic Sub-Account users
```
Comment (bohnertchris) [inline swagger-specs/api.yaml:5167]: Sub account is not a proper noun, do not capitalize. Also, sub account was not previously hyphenated.

File: swagger-specs/api.yaml (hunk 16 above)
Comment (adtanase-adobe) [reply swagger-specs/api.yaml:4701]: This comes from a used library and is like this for all other list objects in CM public API.

File: swagger-specs/api.yaml [hunk 26]
```diff
@@ -4122,6 +4477,240 @@ definitions:
...
+      _links:
+        type: object
+        readOnly: true
+        properties:
+          next:
+            $ref: '#/definitions/HalLink'
+            description: The next page of results.
+          page:
+            $ref: '#/definitions/HalLink'
+            description: An arbitrary page of results.
```
[the remark this answers is not in the document]
Comment (adtanase-adobe) [reply swagger-specs/api.yaml:4698]: This comes from a used library and is like this for all other list objects in CM public API.

File: swagger-specs/api.yaml (hunk 2 above)
Comment (adtanase-adobe) [reply swagger-specs/api.yaml:1398]: Same, it comes from a used library and there are already other 67 previous occurrences in api.yaml of the same text with a full stop.

File: swagger-specs/api.yaml (hunk 4 above)
Comment (adtanase-adobe) [reply swagger-specs/api.yaml:1443]: Same as above

File: swagger-specs/api.yaml (hunk 9 above)
Comment (adtanase-adobe) [reply swagger-specs/api.yaml:1583]: Same as above

File: swagger-specs/api.yaml (hunk 10 above)
Comment (adtanase-adobe) [reply swagger-specs/api.yaml:2732]: Same as above

File: swagger-specs/api.yaml (hunk 14 above)
Comment (adtanase-adobe) [reply swagger-specs/api.yaml:4553]: Same as above, this text is previously used for all lists in CM

File: swagger-specs/api.yaml [hunk 27]
```diff
@@ -4122,6 +4477,240 @@ definitions:
...
+      _links:
+        type: object
+        readOnly: true
+        properties:
+          next:
+            $ref: '#/definitions/HalLink'
+            description: The next page of results.
+          prev:
+            $ref: '#/definitions/HalLink'
+            description: The previous page of results.
```
[the remark this answers is not in the document]
Comment (adtanase-adobe) [reply swagger-specs/api.yaml:4556]: Same as above

File: swagger-specs/api.yaml [hunk 28]
```diff
@@ -4688,64 +5338,64 @@ definitions:
...
         enum:
           - DHGroup24
-          - ECP384
-          - ECP256
-          - DHGroup14
-          - DHGroup2048
-          - DHGroup2
-          - DHGroup1
-          - None
+          - '?ECP384'
```
Comment (cosminseceleanu) [inline swagger-specs/api.yaml:5341]: why this change?
